File: source_funcs/go_transcoder/MINIMUM_DIFFERENCE_MAX_MIN_K_SIZE_SUBSETS/MINIMUM_DIFFERENCE_MAX_MIN_K_SIZE_SUBSETS.go
package main

import (
	"math"
	"sort"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func f_gold(arr []int, N int, K int) int {
	sort.Ints(arr)
	var res int = math.MaxInt64
	for i := int(0); i <= (N - K); i++ {
		var curSeqDiff int = arr[i+K-1] - arr[i]
		res = min(res, curSeqDiff)
	}
	return res
}

File: rust_funcs/go_transcoder/MINIMUM_DIFFERENCE_MAX_MIN_K_SIZE_SUBSETS/MINIMUM_DIFFERENCE_MAX_MIN_K_SIZE_SUBSETS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

fn f_gold(arr: &mut [i32], N: i32, K: i32) -> i32 {
    arr.sort();
    let mut res = std::i32::MAX;
    for i in 0..N-K+1 {
        let cur_seq_diff = arr[i as usize + K as usize - 1] - arr[i as usize];
        res = res.min(cur_seq_diff);
    }
    res
}
